app-libs/exchange-oracle/src/coingecko.rs

```diff
@@ -68,23 +92,25 @@ impl RestPath<String> for CoinGeckoMarket {
 }
 
 impl GetExchangeRate for CoinGeckoClient {
-	fn get_exchange_rate(&mut self, coin: &str, currency: &str) -> Result<f32, Error> {
+	fn get_exchange_rate(&mut self, trading_pair: TradingPair) -> Result<f32, Error> {
+		let curr_id = self.fiat_currency_id(trading_pair.clone())?;
```

Comment:
For the function names you're using `fiat_currency_id` and `crypto_currency_id`, but here only `curr_id`. That's not very consistent. Maybe rename to `fiat_id`?